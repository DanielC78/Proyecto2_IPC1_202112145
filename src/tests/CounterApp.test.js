import CounterApp from "../CounterApp";
import { shallow } from "enzyme";

describe("Prueba de CounterAppp", () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<CounterApp value={10} />);
  });

  test("Debe hacer match con la snapshot", () => {
    const wrapper = shallow(<CounterApp />);
    expect(wrapper).toMatchSnapshot();
  });

  test("Debe mostrar el valor por defecto de 100", () => {
    const wrapper = shallow(<CounterApp value={100} />);

    const counterText = wrapper.find("h2").text().trim();

    expect(counterText).toBe("100");
  });

  test("debe de incrementar en +1", () => {
    wrapper.find("button").at(0).simulate("click");
    const counterText = wrapper.find("h2").text().trim();
    expect(counterText).toBe("11");
  });

  test("debe de reiniciar el valor", () => {
    const valorInicial = 105;
    const wrapper = shallow(<CounterApp value={valorInicial} />);
    wrapper.find("button").at(0).simulate("click");
    wrapper.find("button").at(0).simulate("click");
    wrapper.find("button").at(1).simulate("click");
    const counterText = wrapper.find("h2").text().trim();

    expect(counterText).toBe(`${valorInicial}`);
  });
});
