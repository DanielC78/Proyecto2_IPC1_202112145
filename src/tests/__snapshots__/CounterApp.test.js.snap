// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba de CounterAppp Debe hacer match con la snapshot 1`] = `
<Fragment>
  <h1>
    DBlog
  </h1>
  <h2 />
  <button
    onClick={[Function]}
  >
    +1
  </button>
  <button
    onClick={[Function]}
  >
    Reset
  </button>
  <button
    onClick={[Function]}
  >
    -1
  </button>
</Fragment>
`;
